examples/PyTorch/Inference/BERT/main.py

```diff
@@ -0,0 +1,267 @@
+        },
+        'attention_mask': {
+            0: 'batch_size',
+            1: 'seq_length'
+        },
+        'token_type_ids': {
+            0: 'batch_size',
+            1: 'seq_length'
+        },
+        'last_hidden_state': {
+            0: 'batch_size',
+            1: 'seq_length'
+        },
+        'pooler_output': {
+            0: 'batch_size'
+        }
+    }
+    torch.onnx.export(model,
+                      tuple(inputs),
+                      output_file,
+                      opset_version=10,
+                      input_names=input_names,
+                      output_names=output_names,
+                      dynamic_axes=dynamic_axes)
+
+
+def save_trt_engine(onnx_file: str,
+                    trt_engine_name: str,
+                    opt_shape: list,
+                    min_shape: list = None,
+                    max_shape: list = None,
+                    fp16: bool = False,
+                    int8: bool = False):
+    cmd = f'trtexec --onnx={onnx_file} --saveEngine={trt_engine_name} --workspace=256'
+    type_str = 'fp16' if fp16 else 'fp32'
+    if fp16:
+        cmd += ' --fp16'
+    if int8:
+        cmd += ' --int8'
+    shape_str = f'{opt_shape[0]}x{opt_shape[1]}'
+    cmd += f' --optShapes=input_ids:{shape_str},attention_mask:{shape_str},token_type_ids:{shape_str}'
+    if min_shape:
+        shape_str = f'{min_shape[0]}x{min_shape[1]}'
+        cmd += f' --minShapes=input_ids:{shape_str},attention_mask:{shape_str},token_type_ids:{shape_str}'
+    if max_shape:
+        shape_str = f'{max_shape[0]}x{max_shape[1]}'
+        cmd += f' --maxShapes=input_ids:{shape_str},attention_mask:{shape_str},token_type_ids:{shape_str}'
+    print(cmd)
+    out = os.popen(cmd).read().split('\n')
+    for line in out:
+        print(line)
+
+
+def run_trt_engine(trt_engine_name: str, inputs):
+    import volksdep.converters
+    trt_model = volksdep.converters.load(trt_engine_name)
+
+    # warmup
+    for i in range(20):
+        trt_model(inputs)
+    iters = 100
+    tic = time.time()
+    for i in range(iters):
+        trt_model(inputs)
+    avg_time = (time.time() - tic) / iters
+    print("average time in {} iterations: {} seconds".format(iters, avg_time))
+
+    # profile start
+    _cudart = ctypes.CDLL('/usr/local/cuda/lib64/libcudart.so')
+    res = _cudart.cudaProfilerStart()
+    trt_model(inputs)
+    res = _cudart.cudaProfilerStop()
+    # profile end
+
+
+def blade_trt_optimize(model, inputs, fp16: bool, out_file: str):
+    cfg = torch_blade.Config.get_current_context_or_new().clone()
+    cfg.optimization_pipeline = torch_blade.tensorrt.backend_name()
+    cfg.customize_onnx_opset_version = 12
+    cfg.enable_fp16 = fp16
+
+    traced_model = torch.jit.trace(model.cuda().eval(), inputs,
+                                   strict=False).cuda().eval()
+
+    with cfg, torch_blade.logging.logger_level_context('INFO'):
+        opt_model = torch_blade.optimize(traced_model,
```

Comment:
Nit: Actually, you need to use torch_blade._static_optimize if you want static shape optimization.

Reply:
Fixed. And the number is updated.